compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/PackageResolutionDiagnostic.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.projects.internal;
+
+import io.ballerina.projects.ModuleDescriptor;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectKind;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+
+/**
+ * Decorator for diagnostics exposed in the package resolution phase.
+ *
+ * @since 2.0.0
+ */
+public class PackageResolutionDiagnostic extends PackageDiagnostic {
```

Comment:
Can't we make use of the existing PackageDiagnostic class without introducing a new one?

Reply:
This was like that before, and then in the review meeting, we had advised having a separate class for this one extending  `PackageDiagnostic`.

Reply:
IIRC, during the meeting, the separate diagnostic we discussed was to handle diagnostics that do not have a location.

Reply:
Removed `PackageResolutionDiagnostic` and used `PackageDiagnostic` instead.